src/Host/Client/Impl/rtvs/R/util.R

```diff
@@ -241,4 +245,55 @@ fetch_file <- function(file_path) {
 save_to_project_folder <- function(blob_id, project_name, dest_dir) {
     temp_dir <- paste0(tempdir(), '/RTVSProjects');
     invisible(call_embedded('save_to_project_folder', blob_id, project_name, path.expand(dest_dir), temp_dir));
-}
\ No newline at end of file
+}
+
+autosave_filename <- '~/.Autosave.RData';
+
+query_reload_autosave <- function() {
+	if (!file.exists(autosave_filename)) {
+		return(FALSE);
+	}
+
+	msg <- 'Previous R session terminated unexpectedly due to connectivity issues, and its global workspace has been saved to image "%s". Would you like to reload it?';
+	res <- winDialog('yesno', sprintf(msg, autosave_filename));
+
+	if (identical(res, 'YES')) {
+		# Use try instead of tryCatch, so that any errors are printed as usual.
+		loaded <- FALSE;
+		try({
+			load(autosave_filename, envir = .GlobalEnv);
+			loaded <- TRUE;
+		});
+
+		if (loaded) {
+			message(sprintf('Loaded workspace from autosaved image "%s".\n', autosave_filename));
```

Comment:
Looks like we have a mix of tabs and spaces all over our .R files, and also different indentation levels. I'll go over and reformat them all.